src/index.ts

```diff
@@ -65,38 +105,113 @@ export async function run(): Promise<void> {
     await execute(podman, args);
 
     core.info(`Successfully pushed ${imageToPush} to ${registryPath}.`);
-    core.setOutput('registry-path', registryPath);
+    core.setOutput("registry-path", registryPath);
+
+    // remove the pulled image from the Podman local registry
+    if (isPushingDockerImage) {
+        core.info(`Removing ${imageToPush} from the Podman local registry`);
+        await execute(podman, [ "rmi", imageToPush ]);
+    }
 
     try {
         const digest = (await fs.promises.readFile(digestFile)).toString();
         core.info(digest);
-        core.setOutput('digest', digest);
+        core.setOutput("digest", digest);
     }
     catch (err) {
         core.warning(`Failed to read digest file "${digestFile}": ${err}`);
     }
 }
 
-async function execute(executable: string, args: string[], execOptions: exec.ExecOptions = {}): Promise<{ exitCode: number, stdout: string, stderr: string }> {
+async function pullImageFromDocker(
+    imageName: string,
+    podman: string,
+): Promise<boolean> {
+    try {
+        await execute(podman, [ "pull", `docker-daemon:${imageName}` ]);
+        core.info("Image found and sucessfully pulled from Docker local registry");
+        return true;
+    }
+    catch (err) {
+        core.info("Image not found in Docker local registry");
+        return false;
+    }
+}
+
+async function checkImageInPodman(
+    imageName: string,
+    podman: string,
+): Promise<boolean> {
+    // check if images exist in Podman's local registry
+    core.info("Checking image in Podman local registry");
+    try {
+        await execute(podman, [ "image", "exists", imageName ]);
+        core.info("Image found in Podman local registry");
+    }
+    catch (err) {
+        core.info("Image not found in Podman local registry");
+        core.debug(err);
+        return false;
+    }
+    return true;
+}
+
+async function isPodmanLocalImageLatest(
+    imageName: string,
+    podman: string,
+): Promise<boolean> {
+    // get creation time of the image present in the Podman local registry
+    const podmanLocalImageTimeStamp = await execute(podman, [
+        "image",
+        "inspect",
+        imageName,
+        "--format",
+        "{{.Created}}",
+    ]);
+
+    // get creation time of the image pulled from the Docker local registry
+    // appending 'docker.io/library' infront of image name as pulled image name
+    // from Docker local registry starts with the 'docker.io/library'
+    const pulledImageCreationTimeStamp = await execute(podman, [
+        "image",
+        "inspect",
+        `docker.io/library/${imageName}`,
+        "--format",
+        "{{.Created}}",
+    ]);
+
+    const podmanImageTime = new Date(podmanLocalImageTimeStamp.stdout).getTime();
+
+    const dockerImageTime = new Date(pulledImageCreationTimeStamp.stdout).getTime();
```

Comment:
can you log the timestamps here? the image from \<storage\> was last modified at \<time\> . you can use Date's getLocaleDate and getLocaleTime methods for easy date to-string, though make sure the timezone is in the output since the runners are probably in UTC